### PR TITLE
mannWhitneyUTest + studentTTest + welchTTest + small rankCorr refactor 

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added `mannWitneyUTest`, `studentTTest` and `welchTTest` aggregate functions. Refactored RankCorr a bit.

### CODE REVIEW

File: src/AggregateFunctions/AggregateFunctionMannWhitney.h [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int BAD_ARGUMENTS;
+}
+
+
+struct MannWhitneyData : public StatisticalSample<Float64, Float64>
+{
+    enum class Alternative
+    {
+        TwoSided,
```
Comment (KochetovNicolai) [inline src/AggregateFunctions/AggregateFunctionMannWhitney.h:40]: Comment is needed for each variant.
Probably, `Alternative` is not the best name - alternative in what?

File: tests/queries/0_stateless/01558_ttest_scipy.python [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    test_and_check("studentTTest", rvs1, rvs2, s, p)
+
+    rvs1 = np.round(stats.norm.rvs(loc=0, scale=5,size=500), 5)
+    rvs2 = np.round(stats.norm.rvs(loc=0, scale=5,size=500), 5)
+    s, p = stats.ttest_ind(rvs1, rvs2, equal_var = True)
+    test_and_check("studentTTest", rvs1, rvs2, s, p)
+
+
+    rvs1 = np.round(stats.norm.rvs(loc=0, scale=10,size=65536), 5)
+    rvs2 = np.round(stats.norm.rvs(loc=5, scale=1,size=65536), 5)
```
Comment (KochetovNicolai) [inline tests/queries/0_stateless/01558_ttest_scipy.python:43]: Why variances is different here?
From scipy.stats.ttest_ind docs:
```
This test assumes that the populations have identical variances by default.
```

Comment (KochetovNicolai) [reply tests/queries/0_stateless/01558_ttest_scipy.python:43]: probably, just to check it works the same way

Comment (KochetovNicolai) [reply tests/queries/0_stateless/01558_ttest_scipy.python:43]: Ah, ok, we have `equal_var = True`

Comment (KochetovNicolai) [reply tests/queries/0_stateless/01558_ttest_scipy.python:43]: But it means it is a comparison between Welch’s t-test and Student's?

File: tests/queries/0_stateless/01558_ttest_scipy.python [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    s, p = stats.ttest_ind(rvs1, rvs2, equal_var = True)
+    test_and_check("studentTTest", rvs1, rvs2, s, p)
+
+def test_welch():
+    rvs1 = np.round(stats.norm.rvs(loc=1, scale=15,size=500), 5)
+    rvs2 = np.round(stats.norm.rvs(loc=10, scale=5,size=500), 5)
+    s, p = stats.ttest_ind(rvs1, rvs2, equal_var = True)
+    test_and_check("studentTTest", rvs1, rvs2, s, p)
```
Comment (KochetovNicolai) [inline tests/queries/0_stateless/01558_ttest_scipy.python:51]: Looks like we need to change `studentTTest` -> `welchTTest`?

Comment (nikitamikhaylov) [reply tests/queries/0_stateless/01558_ttest_scipy.python:51]: Sure.

File: tests/queries/0_stateless/01558_ttest_scipy.python (hunk 2 above)
Comment (KochetovNicolai) [reply tests/queries/0_stateless/01558_ttest_scipy.python:43]: no, `equal_var = True` means student

Comment (nikitamikhaylov) [reply tests/queries/0_stateless/01558_ttest_scipy.python:42]: No, it is just to be sure that we have identical answer even in wrong case.

File: src/AggregateFunctions/AggregateFunctionMannWhitney.h (hunk 1 above)
Comment (nikitamikhaylov) [reply src/AggregateFunctions/AggregateFunctionMannWhitney.h:44]: Added a comment.

**Review by KochetovNicolai — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
Added test should be fixed

**Comment by nikitamikhaylov** [comment]
There was a bug in mannWhitney - we have to use std::abs(z) only when alternative is 'two-sided'. Also I noticed that there is significant rounding error in cases of quite big samples (65556), so I reduced the precision on this cases.

**Pull request merged**